### PR TITLE
Smart tomato pet

### PR DESCRIPTION
**Pull request by MrRobDemo**
## About the PR

- Tomato Killer now have a 15% chance to have a ghost role that the player can take
- Smart tomatoes have several phrases, can carry items, and are obliged to serve the owner
- When spraying tomatoes with water, they will heal Brute damage, which will help extend their life
- Some liquids (Blood and RobustHarvest) heal tomatoes better when sprayed
**Water** heals 0.25 Brute
**Blood** heals 0.5 Brust and 0.5 Burn
**RobustHarvest** heals 2 Brust and 2 Burn

## Why / Balance

Tomato Killer are designed to destroy all living things. This PR will allow you to make your tomatoes your pets or servants, who will not kill your colleagues at the first opportunity. Also, the AI ​​of tomatoes does not attack many dangerous creatures, but intelligent tomatoes will be able to protect their owner. Intelligent tomatoes are designed as personal bodyguards or pets, the player will not be able to heal everyone in large quantities. It is also difficult to heal them in battle, because tomatoes run fast, and the owner must be almost right next to them when spraying.

## Technical details

- Added GhostRole, GhostTakeoverAvailable, CanEscapeInventory, Puller, Reactive, ReplacementAccent components to miscellaneous.yml file for tomato
- Added description of tomato role to ghost-role-component.ftl
- Added tomato accent to accents.ftl and full_replacements.yml

## Media

Media1
Media2
Media3
Media4

## Requirements

- [X] I have read and I am following the [Pull Request Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html). I understand that not doing so may get my pr closed at maintainer’s discretion
- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- add: Added 5% chance to make killer tomatoes a ghost role.
- add: Killer tomatoes can now heal from being splashed with water, blood or RobustHarvest.
- tweak: Killer tomatoes can now escape from inventory (only intelligent).

### CODE REVIEW

**Comment by lzk228** [comment]
the last two CL lines can be combined into one

**Comment by baddiePRO** [comment]
Totato

**Comment by Kirill2253** [comment]
Totato

**Comment by Plykiya** [comment]
feels like they should look different? since they're something different than killer tomatos

**Review by TheShuEd — APPROVED** [review]

File: Resources/Prototypes/Entities/Mobs/NPCs/miscellaneous.yml [hunk 1]
```diff
@@ -79,6 +79,57 @@
...
+    needsHands: false
+  - type: Reactive
+    groups:
+      Flammable: [ Touch ]
+      Extinguish: [ Touch ]
+    reactions:
+    - reagents: [ Water, SpaceCleaner ]
+      methods: [ Touch ]
+      effects:
+      - !type:WashCreamPieReaction
```
Comment (slarticodefast) [inline Resources/Prototypes/Entities/Mobs/NPCs/miscellaneous.yml:104]: Is this part needed? As far as I know the tomatos are immune to banana cream pies.

Comment (MrRobDemo) [reply Resources/Prototypes/Entities/Mobs/NPCs/miscellaneous.yml:104]: Perhaps, but the component does not work for me without this part. Perhaps there is a way to do without it

Comment (slarticodefast) [reply Resources/Prototypes/Entities/Mobs/NPCs/miscellaneous.yml:104]: I just gave it a test and it works fine. Remove lines 101 to 104.

File: Resources/Locale/en-US/ghost/roles/ghost-role-component.ftl [hunk 2]
```diff
@@ -306,3 +306,10 @@ ghost-role-information-pirate-captain-description = Argh matey! You are in charg
 
 ghost-role-information-artifact-name = Sentient Artifact
 ghost-role-information-artifact-description = Enact your eldritch whims. Forcibly activate your nodes for good or for evil.
+
+ghost-role-information-syndie-assaultborg-name = Syndicate Assault Borg
+ghost-role-information-syndie-assaultborg-description = Nuclear operatives needs reinforcements. You, a cold silicon killing machine, will help them. More dakka!
```
Comment (slarticodefast) [inline Resources/Locale/en-US/ghost/roles/ghost-role-component.ftl:311]:
```suggestion
```
Remove this, this string was renamed and moved in #28098

File: Resources/Prototypes/Accents/full_replacements.yml [hunk 3]
```diff
@@ -174,3 +174,12 @@
   - accent-words-nymph-2
   - accent-words-nymph-3
   - accent-words-nymph-4
+
+- type: accent
+  id: tomatokiller
```
Comment (slarticodefast) [inline Resources/Prototypes/Accents/full_replacements.yml:179]:
```suggestion
  id: tomatoKiller
```

File: Resources/Prototypes/Entities/Mobs/NPCs/miscellaneous.yml [hunk 4]
```diff
@@ -79,6 +79,53 @@
...
+      methods: [ Touch ]
+      effects:
+      - !type:HealthChange
+        scaleByQuantity: true
+        damage:
+          groups:
+            Brute: -2
+            Burn: -2
+  - type: ReplacementAccent
+    accent: tomatokiller
```
Comment (slarticodefast) [inline Resources/Prototypes/Entities/Mobs/NPCs/miscellaneous.yml:128]:
```suggestion
    accent: tomatoKiller
```

**Review by slarticodefast — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/Entities/Mobs/NPCs/miscellaneous.yml [hunk 5]
```diff
@@ -79,6 +79,53 @@
   name: tomato killer
   description: Looks like it's not you eating tomatoes today, it's the tomatoes eating you.
   components:
+  - type: GhostRole
+    prob: 0.15
```
Comment (slarticodefast) [inline Resources/Prototypes/Entities/Mobs/NPCs/miscellaneous.yml:83]:
```suggestion
    prob: 0.05
```
Should be a little lower considering how many tomatos can be harvested in a short amount of time. We don't want to spam ghost roles.

File: Resources/Prototypes/Entities/Mobs/NPCs/miscellaneous.yml [hunk 6]
```diff
@@ -79,6 +79,53 @@
...
+    name: ghost-role-information-tomatokiller-name
+    description: ghost-role-information-tomatokiller-description
+    rules: ghost-role-information-familiar-rules
+    raffle:
+      settings: short
+  - type: GhostTakeoverAvailable
+  - type: CanEscapeInventory
+    baseResistTime: 3
+  - type: Puller
+    needsHands: false
```
Comment (slarticodefast) [inline Resources/Prototypes/Entities/Mobs/NPCs/miscellaneous.yml:95]:
```suggestion
```
Ok, one last thing that came up. Please remove the ability for pulling, as other small mobs without hands like mice or slimes cannot do this either. Once that is done this should be good to merge.

**Review by slarticodefast — APPROVED** [review]

**Comment by slarticodefast** [comment]
Thank you for your contribution!

**Pull request merged**